relay-event-normalization/src/nel.rs

```diff
@@ -0,0 +1,77 @@
+//! Contains normalization for NEL reports.
+
+use chrono::{DateTime, Duration, Utc};
+use relay_event_schema::protocol::{
+    Contexts, Event, HeaderName, HeaderValue, Headers, LogEntry, NelContext, NetworkReportRaw,
+    PairList, Request, ResponseContext, Timestamp,
+};
+use relay_protocol::Annotated;
+
+/// Normalizes the event using the provided [`NetworkReportRaw`].
+pub fn normalize(event: &mut Event, nel: Annotated<NetworkReportRaw>) {
+    // If the incoming NEL report is empty or it contains an empty body, just exit.
+    let Some(nel) = nel.value() else {
+        return;
+    };
+    let Some(body) = nel.body.value() else {
+        return;
+    };
+
+    event.logentry = Annotated::new(LogEntry::from({
+        if nel.ty.value().map_or("", |v| v.as_str()) == "http.error" {
+            format!(
+                "{} / {} ({})",
+                body.phase.as_str().unwrap_or(""),
```

Comment:
Not sure if this happens in the wild, but should we write some placeholder here?
```suggestion
                body.phase.as_str().unwrap_or("<unknown-phase>"),
```